inference-engine/src/mkldnn_plugin/utils/blob_dump.cpp

```diff
@@ -73,177 +76,131 @@ static TensorDesc parse_header(IEB_HEADER &header) {
-    if (!_blob)
-        IE_THROW() << "Dumper cannot dump empty Blob";
-
-    if (_blob->buffer().as<float*>() == nullptr)
-        IE_THROW() << "Dumper cannot dump. Blob is not allocated.";
+    if (memory == nullptr)
+        IE_THROW() << "Dumper cannot dump. Memory is not allocated.";
 
-    IEB_HEADER header = prepare_header(_blob->getTensorDesc());
-    Blob::Ptr pln_blob = prepare_plain_data(_blob);
+    IEB_HEADER header = prepare_header(memory->GetDescWithType<MKLDNNMemoryDesc>());
+    std::vector<uint8_t> data;
+    prepare_plain_data(this->memory, data);
 
     header.data_offset = sizeof(header);
-    header.data_size = pln_blob->byteSize();
+    header.data_size = data.size();
     header.scaling_data_offset = 0;
     header.scaling_data_size = 0;
 
-    if (_scales) {
-        header.scaling_axis = 1;
-        header.scaling_data_offset = header.data_offset + header.data_size;
-        header.scaling_data_size = _scales->byteSize();
-    }
-
-    stream.write(reinterpret_cast<char*>(&header), sizeof(header));
-    stream.write(pln_blob->buffer().as<char*>(), pln_blob->byteSize());
-
-    if (_scales) {
-        stream.write(_scales->buffer().as<char*>(), _scales->byteSize());
-    }
+    stream.write(reinterpret_cast<const char*>(&header), sizeof(header));
+    stream.write(reinterpret_cast<char*>(data.data()), data.size());
 }
 
 void BlobDumper::dumpAsTxt(std::ostream &stream) const {
-    if (!_blob)
-        IE_THROW() << "Dumper cannot dump empty Blob";
-
-    if (_blob->buffer().as<float*>() == nullptr)
-        IE_THROW() << "Dumper cannot dump. Blob is not allocated.";
+    if (memory == nullptr)
+        IE_THROW() << "Dumper cannot dump. Memory is not allocated.";
 
-    SizeVector dims = _blob->getTensorDesc().getDims();
+    const auto dims = memory->GetDims();
+    size_t data_size = product(dims);
 
     // Header like "U8 4D shape: 2 3 224 224 ()
-    stream << _blob->getTensorDesc().getPrecision().name() << " "
+    stream << memory->GetDesc().getPrecision().name() << " "
            << dims.size() << "D "
            << "shape: ";
     for (size_t d : dims) stream << d << " ";
-    stream << "(" << _blob->size() << ")" <<
-    " by address 0x" << std::hex << _blob->buffer().as<long long>() << std::dec <<std::endl;
-
-    // Dump data
-    // TODO [DS]: blob dumper should be rewritten using Memory object
-    MKLDNNMemoryDesc mdesc = MemoryDescUtils::convertToMKLDNNMemoryDesc(_blob->getTensorDesc());
-    mkldnn::memory::desc desc = mdesc;
-    mkldnn::impl::memory_desc_wrapper blob_wrp(desc.data);
-
-    size_t data_size = _blob->size();
-    switch (_blob->getTensorDesc().getPrecision()) {
-        case Precision::FP32: {
-            auto *blob_ptr = _blob->buffer().as<float*>();
+    stream << "(" << data_size << ")" <<
+    " by address 0x" << std::hex << reinterpret_cast<const long long *>(memory->GetData()) << std::dec <<std::endl;
+
+    const void *ptr = memory->GetData();
```

Comment:
M.b. `GetPtr()`?

Reply:
bellow we used ```getOffset``` to compute physical offset, this method already takes into account offset0. But ```GetPtr``` return ```data_ptr + offset0```. so i guess ```GetData``` valid here

Reply:
Nope. `getOffset()` gives you linear index only without initial offset to the real data.